.github/workflows/main.yml

```diff
@@ -4,11 +4,16 @@ on:
   push:
 
 jobs:
+  build-matrix:
+    name: Generate Build Matrix
+    uses: ./.github/workflows/matrix-python.yml
+
   test-exaslct-alias-script:
+    needs: build-matrix
     strategy:
       fail-fast: false
-      matrix:
-        python_version: ["3.10", "3.11", "3.12"]
+      matrix: ${{ fromJson(needs.build-matrix.outputs.matrix) }}
+
     runs-on: ubuntu-latest
```

Comment:
pin

Reply:
@Nicoretti what is your opinion about that?
The generated GH workflows from the toolbox use all
```
 matrix: ${{ fromJson(needs.build-matrix.outputs.matrix) }}
```

@tkilias , actually, the Python versions are pinned in `./.github/workflows/matrix-python.yml` which is also checked in. I believe the idea is just to use the same Python versions accross all GH workflows (wherever you want to check against different versions of Python).

Reply:
it is not about python versions, it is about ubuntu versions

Reply:
Ah, sorry, got it.